jax/_src/lax/slicing.py

```diff
@@ -1244,15 +1249,17 @@ def _gather_fill(operand, indices, *, dimension_numbers, slice_sizes,
 
 def _gather_jvp_rule(g, operand, indices, *, dimension_numbers,
                      slice_sizes, unique_indices, indices_are_sorted, mode,
-                     fill_value):
+                     fill_value, check):
+  del check
   return gather(g, indices, dimension_numbers, slice_sizes,
                 unique_indices=unique_indices,
                 indices_are_sorted=indices_are_sorted, mode=mode,
                 fill_value=0)
```

Comment:
Why don't you add a `check=check` here?

Reply:
I think this rule should call gather_p.join(...) and pass in check explicitly. That way we aren't relying on any global state.

If there's retracing happening then the context manager won't be active anymore so it's safer to just be explicit here.

Reply:
actually: wouldn't this cause issues when rules produce different primitives?

eg. `gather_transpose_rule` calls `scatter_add`. In this case `gather` and `scatter` share the same `check` set (`index_checks` checks both `gather`s and `scatter`s), but some primitives wouldn't. 

I guess the `check` arg cannot be a boolean, and needs to carry along the set of all active checks: then the primitive can be decomposed into different primitives by the transformation rules and those primitives will know if they need to be checked.

I also just realized that some (few) primitives can have multiple checks (eg. `scatter_add` has `index` and `nan` checks.

Reply:
If a primitive in the fwd pass passes all its checks, should its primitives in the bwd pass also pass the checks?

Reply:
no don't think so, this is one of the examples in our documentation that only fails on the bwd (there's also no `mul` on the fwd)

```
def f(x):
 return x / (1 + jnp.sqrt(x))

grad_f = jax.grad(f)

err, _ = checkify.checkify(grad_f, errors=checkify.nan_checks)(0.)
print(err.get())
>> nan generated by primitive mul at <...>:3 (f)
```